tests/bgp/test_bgp_speaker.py

```diff
@@ -246,7 +246,7 @@ def bgp_speaker_announce_routes_common(common_setup_teardown,
     announce_route(ptfip, lo_addr, peer_range, vlan_ips[0].ip, port_num[2])
 
     logger.info("Wait some time to make sure routes announced to dynamic bgp neighbors")
-    time.sleep(30)
+    time.sleep(90)
```

Comment:
why not 'wait_until' or retry? sleep is inefficient.

Reply:
I tried to call  duthost.bgp_facts() in a loop - it raises error when calling second time or more

Reply:
Does this method help? https://github.com/Azure/sonic-mgmt/blob/master/tests/common/helpers/snmp_helpers.py#L31

Reply:
> I tried to call duthost.bgp_facts() in a loop - it raises error when calling second time or more

can you investigate why, use wait_until makes more sense

Reply:
I agree that use `wait_until` is more efficient. If calling `duthost.bgp_facts()` raises error, you can call it with argument `module_ignore_erros=True`, then explicitly check the returned result. Timeout of wait_until can be set to 90 seconds.